src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -968,6 +968,12 @@ FixedwingPositionControl::control_auto_fixed_bank_alt_hold(const float control_i
 		_att_sp.thrust_body[0] = min(get_tecs_thrust(), _param_fw_thr_max.get());
 	}
 
+	// Special case: if z or vz estimate is invalid we cannot control height anymore. To prevent a
+	// "climb-away" we set the thrust to 0 in that case.
+	if (!_local_pos.z_valid || !_local_pos.v_z_valid) {
+		_att_sp.thrust_body[0] = 0.f;
+	}
```

Comment:
May be the thrust_body field should be an else case of this condition?

Reply:
I think I prefer it separated. Otherwise we could end up in a funny situation where the vehicle descends with throttle=0 but then enables throttle again when on the ground when landed (in case FW_THR_MIN is not 0).

Reply:
Wait, no the logic would not change if you do it as an else case no?

Reply:
yeah I now adapted it now slightly, also set it to MIN like in Landed case.